toolchain/check/import_ref.cpp

```diff
@@ -371,6 +371,31 @@ class ImportRefResolver {
     return GetLocalConstantId(import_ir_.types().GetInstId(type_id));
   }
 
+  // Returns the ConstantId for a TypeId. Adds unresolved constants to
+  // work_stack_.
+  auto GetLocalCanonicalInstBlockId(SemIR::InstBlockId import_block_id)
+      -> SemIR::InstBlockId {
+    if (!import_block_id.is_valid()) {
+      return SemIR::InstBlockId::Invalid;
+    }
+
+    auto initial_work = work_stack_.size();
+
+    // Import all the values in the block.
+    auto import_block = import_ir_.inst_blocks().Get(import_block_id);
+    llvm::SmallVector<SemIR::InstId> inst_ids;
+    inst_ids.reserve(import_block.size());
+    for (auto import_inst_id : import_block) {
+      auto const_id = GetLocalConstantId(import_inst_id);
+      inst_ids.push_back(const_id.inst_id());
+    }
+
+    if (HasNewWork(initial_work)) {
```

Comment:
We typically use HasNewWork in helpers, so suggesting a tiny comment to call it out.
```suggestion
    // Don't canonicalize the block when it's incomplete.
    if (HasNewWork(initial_work)) {
```

Note though, not typically doing this may be less than ideal -- it seems like `GetLocalParamRefsId` should maybe do similar, rather than adding an incomplete block, for example.

Reply:
Sent #4036

Reply:
Nevermind, realized it's not needed elsewhere because constants are fetched first. They're written not to do the duplicative work, differently.

That may be something to consider here, though... Maybe it should return the vector, and only add the canonical block if everything's ready? Otherwise, you could end up canonicalizing when a later constant is incomplete. You may still end up with the same ID, but it's still duplicative work.

Reply:
Changed to return the vector as suggested.